projecttacoma/fhir-bundle-calculator
# fqm-execution integration

# Summary

Add the ability to run calculation using fqm-execution within the CLI.

## New behavior

New CLI options/calculation behavior. For the options:

* If `-t` is `"fqm"` or is omitted, then `-b` is required, and `-u` and `-m` are irrelevant and forbidden
* If `-t` is `"http"`, then `-u` and `-m` are required  and `-b` is forbidden

## Code changes

* Dependencies
  * Add fqm-execution
  * Update axios to get rid of security vulnerability
  * Upgrade commander version
* Update README to reflect new functionality
* New Functionality
  * Modify calculation.js logic to handle measure reports independent of where they come from
  * Add logic to CLI to get measure report from either cqf ruler or fqm-exec based on which type is specified
  * Move http interactions to helper file for simplicity
* Testing
  * Simplify calculation.test.js as the tests no longer require http operations, just measure reports
  * Add tests for new httpUtils that mock web requests
* Misc
  * Remove uses of `cqf-ruler` nomenclature. Theoretically it could be any server that supports measure evaluation
  * Bump major version (breaking changes)

# Testing guidance

* Test different combinations of provided flags to ensure that the logic above holds
* Run the cli with a loaded cqf-ruler instance, and ensure that results are unchanged
* Run the cli with fqm type providing the measure bundle, ensure it behaves the same as if it were cqf ruler but 10x faster.
